src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +53,127 @@
-      System.out.println(lineStart + " " + lineDuration + " "
-          + " " + lineContent);
+      datastore.put(createLineEntity(node, lectureId));
     }
   }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PreparedQuery preparedQuery = getLectureTranscriptQuery(request);
+    ImmutableList<TranscriptLine> transcriptLines = getTranscriptLines(preparedQuery);
+    writeResponseAsJson(response, transcriptLines);
+  }
+
+  /**
+   * Returns the query for the lecture transcripts based on lecture id indicated in {@code request}.
+   */
+  private PreparedQuery getLectureTranscriptQuery(HttpServletRequest request) {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    Key lecture = KeyFactory.createKey(PARAM_LECTURE, lectureId);
+
+    Filter lectureFilter =
+        new FilterPredicate(TranscriptLine.PROP_LECTURE, FilterOperator.EQUAL, lecture);
+
+    Query query = new Query(TranscriptLine.ENTITY_KIND)
+                      .setFilter(lectureFilter)
+                      .addSort(TranscriptLine.PROP_START, SortDirection.ASCENDING);
+    return datastore.prepare(query);
+  }
+
+  private ImmutableList<TranscriptLine> getTranscriptLines(PreparedQuery preparedQuery) {
+    ImmutableList.Builder<TranscriptLine> lineBuilder = new ImmutableList.Builder<>();
+    for (Entity entity : preparedQuery.asQueryResultIterable()) {
+      lineBuilder.add(TranscriptLine.fromLineEntity(entity));
+    }
+    return lineBuilder.build();
+  }
+
+  /**
+   * Writes {@code list} as Json to {@code response}.
+   */
+  private void writeResponseAsJson(HttpServletResponse response, ImmutableList<TranscriptLine> list)
+      throws IOException {
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(list));
+  }
+
+  /**
+   * Creates a line entity using the attributes from {@code node} and {@code lectureId}.
+   */
+  private Entity createLineEntity(Node node, long lectureId) {
+    Element element = (Element) node;
+    String lineContent = node.getTextContent();
+    String lineStart = element.getAttribute(ATTR_START);
+    String lineDuration = element.getAttribute(ATTR_DURATION);
+
+    Entity lineEntity = new Entity(TranscriptLine.ENTITY_KIND);
+    // TODO: Change PARAM_LECTURE to Lecture.ENTITY_KIND once lectureServlet is
+    // merged to this branch.
+    lineEntity.setProperty(
+        TranscriptLine.PROP_LECTURE, KeyFactory.createKey(PARAM_LECTURE, lectureId));
+    lineEntity.setProperty(TranscriptLine.PROP_CONTENT, lineContent);
+    lineEntity.setProperty(TranscriptLine.PROP_START, lineStart);
+    lineEntity.setProperty(TranscriptLine.PROP_DURATION, lineDuration);
+    return lineEntity;
+  }
+
+  @VisibleForTesting
+  protected void init(DatastoreService testDatastore) {
+    datastore = testDatastore;
+  }
```

Comment:
Why do we pass a "testDatastore" here please?

Reply:
To check that doPost() actually puts the entities in datastore, I need to have access to it. I'll add a comment for it :)